flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -21,18 +26,98 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
         options.logger(SentryLevel.debug,
             'Scheduler binding is null. Can\'t auto detect app start time.');
       } else {
-        schedulerBinding.addPostFrameCallback((timeStamp) {
+        schedulerBinding.addPostFrameCallback((timeStamp) async {
           // ignore: invalid_use_of_internal_member
-          _native.appStartEnd = options.clock();
+          final appStartEnd = options.clock();
+          _native.appStartEnd = appStartEnd;
+
+          if (!_native.didFetchAppStart) {
+            final nativeAppStart = await _native.fetchNativeAppStart();
+            final measurement = nativeAppStart?.toMeasurement(appStartEnd!);
+
+            // We filter out app start more than 60s.
+            // This could be due to many different reasons.
+            // If you do the manual init and init the SDK too late and it does not
+            // compute the app start end in the very first Screen.
+            // If the process starts but the App isn't in the foreground.
+            // If the system forked the process earlier to accelerate the app start.
+            // And some unknown reasons that could not be reproduced.
+            // We've seen app starts with hours, days and even months.
+            if (nativeAppStart == null ||
+                measurement == null ||
+                measurement.value >= 60000) {
+              _appStartTracker?.setAppStartInfo(null);
+              return;
+            }
+
+            final appStartInfo = AppStartInfo(
+              DateTime.fromMillisecondsSinceEpoch(
+                  nativeAppStart.appStartTime.toInt()),
+              appStartEnd,
+              measurement,
+            );
+
+            _appStartTracker?.setAppStartInfo(appStartInfo);
+          } else {
+            _appStartTracker?.setAppStartInfo(null);
+          }
```

Comment:
`l`: You could turn this else branch into an early return above to reduce the nesting. Generally speaking, this code is pretty nested, and you could apply early return also in the code you didn't touch here to reduce the nesting, but I would do that in an extra PR.